source/extensions/http/cache/file_system_http_cache/cache_file_fixed_block.h

```diff
@@ -0,0 +1,121 @@
+#pragma once
+
+#include <sys/types.h>
+
+#include <cstddef>
+#include <cstdint>
+
+#include "source/common/common/byte_order.h"
+
+#include "absl/strings/string_view.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Cache {
+namespace FileSystemHttpCache {
+
+// This represents a minimal header block on the cache entry file; it uses a struct that
+// is required to be packed, and explicit network byte order, to ensure consistency.
+class CacheFileFixedBlock {
+public:
+  CacheFileFixedBlock();
+
+  // Deserializes the string representation of a CacheFileFixedBlock into this instance.
+  void populateFromStringView(absl::string_view s);
+
+  // Returns the size in bytes of a CacheFileFixedBlock.
+  static size_t size() { return sizeof(contents_); }
+
+  // fileId is a fixed value used to identify that this is a cache file.
+  uint32_t fileId() const { return fromEndianness<ByteOrder::BigEndian>(contents_.file_id_); }
```

Comment:
Sure, done. Slightly weird because typically network byte order is the "consistency" one, but if we care about a single instruction of performance in the context of large disk operations then the optimization is worthwhile.